.teamcity/src/main/kotlin/model/FunctionalTestBucketProvider.kt

```diff
@@ -257,21 +174,33 @@ class SmallSubprojectBucket(val subprojectsBuildTime: List<SubprojectTestClassTi
         }
 
     override fun createFunctionalTestsFor(model: CIBuildModel, stage: Stage, testCoverage: TestCoverage, bucketIndex: Int): FunctionalTest =
-        FunctionalTest(model,
+        FunctionalTest(
+            model,
             getUuid(model, testCoverage, bucketIndex),
             getName(testCoverage),
             getDescription(testCoverage),
             testCoverage,
             stage,
-            subprojects.map { it.name }
+            subprojects.map { it.name },
+            enableTestDistribution
         )
 
     override fun getName(testCoverage: TestCoverage) = truncateName("${testCoverage.asName()} (${subprojects.joinToString(",") { it.name }})")
 
     override fun getDescription(testCoverage: TestCoverage) = "${testCoverage.asName()} for ${subprojects.joinToString(", ") { it.name }}"
 }
 
-class TestClassTime(var testClass: String, val sourceSet: String, var buildTimeMs: Int) {
+class TestClassTime(
+    var testClass: String,
+    val sourceSet: String,
+    var buildTimeMs: Int
```

Comment:
I think we should move this class to the Generator, and then have something else here which doesn't need the ms time.

Reply:
Done.